src/planning/include/planning/smoothing2.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef SRC_PLANNING_INCLUDE_PLANNING_SMOOTHING2_HPP_
+#define SRC_PLANNING_INCLUDE_PLANNING_SMOOTHING2_HPP_
+
+#include "common_lib/structures/path_point.hpp"
+#include "common_lib/structures/pose.hpp"
+#include "utils/smoothing_config.hpp"
+#include "utils/splines.hpp"
+
+using PathPoint = common_lib::structures::PathPoint;
+using Pose = common_lib::structures::Pose;
+
+/**
+ * @brief class that defines the path smoothing algorithm
+ *
+ */
+class PathSmoothing {
```

Comment:
The config and class name should be consistent. I'd pick one out of "Smoothing" or "PathSmoothing" and maintain the same entity name across the different usages